### PR TITLE
[MRG] Update contributing guide

### PR DESCRIPTION
**Pull request by betatim**
I wanted to work on kubespawner and first couldn't find the contributing guide because I looked in `CONTRIBUTING.md` which was empty. I've edited `CONTRIBUTING.md` to be similar to what we use for repo2docker https://repo2docker.readthedocs.io/en/latest/contributing/contributing.html and contain the contents of `SETUP.md`.

Once I found `SETUP.md` I had to make a few changes to the `jupyterhub_config.py` to get things to work locally (with jhub 1.1). These changes are included in this PR but I am unsure if they are needed/correct.

What do you think of these changes/re-organisation?

### CODE REVIEW

**Comment by consideRatio** [comment]
It is a bit tricky for me to spot what you updated among what was just moved. Can you point it out with code review comments perhaps?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,189 @@
...
+    pods inside minikube.
+
+    ```bash
+    # Linux
+    sudo ip route add 172.17.0.0/16 via $(minikube ip)
+
+    # MACOS
+    sudo route -n add -net 172.17.0.0/16 $(minikube ip)
+    # later on you can undo this it with
+    sudo route delete -net 172.17.0.0
```
Comment (betatim) [inline CONTRIBUTING.md]: added this command

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,189 @@
...
+   ```sh
+   git clone https://github.com/jupyterhub/kubespawner.git
+   ```
+
+1. Setup a virtual environment. After cloning the repository, you should set up an
+   isolated environment to install libraries required for running / developing
+   kubespawner.
+
+   There are many ways of doing this: conda envs, virtualenv, pipenv, etc. Pick
+   your favourite. We show you how to use venv:
```
Comment (betatim) [inline CONTRIBUTING.md]: New sentence

File: jupyterhub_config.py [hunk 3]
```diff
@@ -4,8 +4,8 @@
 
 c.JupyterHub.spawner_class = 'kubespawner.KubeSpawner'
```
Comment (betatim) [inline jupyterhub_config.py]: All changes in this file are new

File: CONTRIBUTING.md (hunk 1 above)
Comment (betatim) [reply CONTRIBUTING.md]: if someone knows what the linux equivalent is we can add that above as well

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,3 +1,189 @@
...
+We use Sphinx to build the kubespawner documentation. You can make changes to
+the documentation with any text editor and directly through the GitHub website.
+
+For small changes (like typos) you do not need to setup anything locally. For
+larger changes we recommend you build the documentation locally so you can see
+the end product in its full glory.
+
+To make edits through the GitHub website visit https://github.com/jupyterhub/kubespawner/tree/master/docs/source, open the file you would like to edit and then click "edit". GitHub will
+walk you through the process of proposing your change ("making a Pull Request").
+
```
Comment (betatim) [inline CONTRIBUTING.md]: everything above here is "new" in the sense that it didn't use to be anywhere in the kubespawner repo. it is "old" in the sense that it is copied more or less directly from repo2docker's contrib guide.

File: CONTRIBUTING.md (hunk 1 above)
Comment (GeorgianaElena) [reply CONTRIBUTING.md]: The man page says:
```
ip route { add | del | change | append | replace } ROUTE
```
So, I believe the "undo" linux equivalent would be
```
sudo ip route del 172.17.0.0/16
```

Comment (betatim) [reply CONTRIBUTING.md]: Thanks for finding out, I added it.

**Comment by consideRatio** [comment]
This LGTM! Thank you @betatim!

**Pull request merged**